tests/query_workflow.go

```diff
@@ -146,26 +145,72 @@ func (s *ClientFunctionalSuite) TestQueryWorkflow_Consistent_PiggybackQuery() {
 }
 
 func (s *ClientFunctionalSuite) TestQueryWorkflow_QueryWhileBackoff() {
-	testname := s.T().Name()
-	workflowFn := func(ctx workflow.Context) (string, error) {
-		workflow.SetQueryHandler(ctx, testname, func() (string, error) {
+
+	testnameFail := "backoff_query_fail"
+	testnamePass := "backoff_query_pass"
+
+	workflowFnFail := func(ctx workflow.Context) (string, error) {
+		_ = workflow.SetQueryHandler(ctx, testnameFail, func() (string, error) {
 			return "should-reach-here", nil
 		})
-		return "", temporal.NewApplicationError("retry-me", "test-error")
+		return "", nil
+	}
+	workflowFnPass := func(ctx workflow.Context) (string, error) {
+		_ = workflow.SetQueryHandler(ctx, testnamePass, func() (string, error) {
+			return "should-reach-here", nil
+		})
+		return "", nil
```

Comment:
Why do you need different functions? Also why different query handler names?

Reply:
without it engine panics with "function with the same name is already registered".